Source/astcenc_ideal_endpoints_and_weights.cpp

```diff
@@ -1115,48 +1116,39 @@ void compute_ideal_quantized_weights_for_decimation_table(
 
 	int i = 0;
 
-#if ASTCENC_AVX >= 2
-	//  TODO: This is currently 4-wide. Could try 8?
-	int clipped_weight_count = weight_count & ~3;
-	__m128i shuf = _mm_set_epi8(-1, -1, -1, -1, -1, -1, -1, -1,
-	                            -1, -1, -1, -1, 12,  8,  4,  0);
-	__m128 scalev = _mm_set1_ps(scale);
-	__m128 scaled_low_boundv = _mm_set1_ps(scaled_low_bound);
-	for (/* Loop vector */; i < clipped_weight_count; i += 4)
+#if ASTCENC_SIMD_WIDTH > 1
+	// SIMD loop; process weights in SIMD width batches while we can.
+	int clipped_weight_count = weight_count & ~(ASTCENC_SIMD_WIDTH-1);
+	vfloat scalev(scale);
+	vfloat scaled_low_boundv(scaled_low_bound);
+	vfloat quant_level_m1v(quant_level_m1);
+	vfloat rscalev(rscale);
+	vfloat low_boundv(low_bound);
+	for (/*Vector loop */; i < clipped_weight_count; i += ASTCENC_SIMD_WIDTH)
 	{
-		__m128 ix = _mm_load_ps(&weight_set_in[i]);
-		ix = _mm_mul_ps(ix, scalev);
-		ix = _mm_sub_ps(ix, scaled_low_boundv);
+		vfloat ix = loada(&weight_set_in[i]) * scalev - scaled_low_boundv;
```

Comment:
The data storage for `weight_set_in` and friends are allocated as part of the `compress_fixed_partition_buffers` scratch buffer struct allocated for each thread. We allocate the structure with `alignas()`, but we don't tag the individual members so it's possible they could unalign in future if we change that structure. We should add alignas() to that struct for all of the individual members (at least the bottom 4 entries).